feat: sign in UI update

imblue-dabadee opened the pull request:
# Sign-In UI update
Redesigns the existing Sign-In UI.
Notable changes:
- Changes the visuals for the `ConfirmSimulationButton` component.
- Removes header for Personal Sign.

## Previous
![Screenshot 2023-05-03 at 1 15 01 pm](https://user-images.githubusercontent.com/106779544/236007343-6918f39a-6c4a-4260-a43d-18072fa2c72b.png)


## New
<img width="409" alt="Screenshot 2023-05-03 at 1 12 25 pm" src="https://user-images.githubusercontent.com/106779544/236006805-ca3582b2-1f4f-4641-bf4c-f2d263f333d7.png">

mindofmar committed:
get popup to show on bypass

mindofmar committed:
implement simulation into bypass protection

mindofmar committed:
working bypass protection on eth_sendtransaction

mindofmar committed:
refactor popup components to use current simulation

mindofmar committed:
add warning banner for bypass

mindofmar committed:
add support for signed and unsigned methods

mindofmar committed:
generate request id so that we dont run twice

mindofmar committed:
revert some things to try and get this working

mindofmar committed:
keep debugging comparison

mindofmar committed:
working solution for skipping arbitrum txns

mindofmar committed:
compare objects correctly to prevent bypass popup after confirming transaction

mindofmar committed:
refactor bypass message to append to current warnings

mindofmar committed:
minor fixes

mindofmar committed:
remove redundant bypassed field

mindofmar committed:
remove commented code

mindofmar committed:
fix: bypasses were being added to approved txns due to poor typing

imblue-dabadee committed:
feat: new personal sign screen

imblue-dabadee committed:
refactor: popup has personal sign

imblue-dabadee committed:
chore: add new assets

imblue-dabadee committed:
style: css

imblue-dabadee committed:
refactor: componentize

imblue-dabadee committed:
fix: import personalSign

imblue-dabadee committed:
fix: styling for p more specific

imblue-dabadee committed:
style: scrollbar

imblue-dabadee committed:
fix: size of reject

imblue-dabadee committed:
fix: loading screen with skip

imblue-dabadee committed:
chore: add arbitrum asset

imblue-dabadee committed:
chore: update package-lock.json

imblue-dabadee committed:
feat: add new colours

imblue-dabadee committed:
fix: origin comes from args instead of result

imblue-dabadee committed:
chore: remove redundant code

mindofmar commented on src/styles/simulation/ConfirmSimulationButton.module.css:
```diff
@@ -0,0 +1,25 @@
+    align-items: center;
+    gap: 10px;
+    margin: 10px 0;
+    width: 183px;
```
previously we used `col-6` for this I think right? Not sure if I like using a hard coded width here instead of `col-6`

mindofmar commented on src/components/simulation/DappLogoWithChain.tsx:
```diff
@@ -0,0 +1,166 @@
+    switch (color) {
+      case 'pink':
+        setBackgroundGradient1(
+          'radial-gradient(88.3% 88.3% at 50% 95%, rgba(255, 0, 122, 0.5) 0%, rgba(255, 0, 122, 0) 100%)'
```
too much css inside this JSX file

mindofmar commented on src/components/simulation/DappLogoWithChain.tsx:
```diff
@@ -0,0 +1,166 @@
+          }}
+        ></div>
+        <div
+          style={{
```
you have a lot of repetitive CSS here. Use a class

mindofmar commented on src/styles/simulation/PersonalSign/PersonalSign.module.css:
```diff
@@ -0,0 +1,54 @@
+    color: #636363;
+}
+
+.df-ac-g5 {
```
what does `df-ac-g5` mean???

mindofmar commented on src/components/simulation/PersonalSign.tsx:
```diff
@@ -0,0 +1,87 @@
+  // Fetch the metadata json
+  useEffect(() => {
+    const fetchData = async () => {
+      const response = await fetch('https://walletguard-public-prod.s3.us-east-2.amazonaws.com/url/metadata/'+ simulation.args.origin + '.json');
```
set this URL in `environment.ts`

mindofmar commented on src/components/simulation/PersonalSign.tsx:
```diff
@@ -0,0 +1,87 @@
+  // Set the chain logo path
+  useEffect(() => {
+    switch (simulation.args.chainId) {
+      case '42161':
```
does there need to be a case here for `eth`? I understand that's the default, but if it ever needed to be switched in the other direction from arb => eth, this code would not cover that path

mindofmar commented on src/components/simulation/PersonalSign.tsx:
```diff
@@ -0,0 +1,87 @@
+      <DappLogoWithChain
+        name={metadata.name}
+        logoPath={metadata.logo}
+        chainLogoPath="/images/asset_logos/eth-mainnet.png"
```
I think you hard coded this and forgot to replace it- this should be `chainLogoPath` right?

imblue-dabadee committed:
refactor: styling for DappLogoWithChain

imblue-dabadee committed:
refactor: env for S3 URL

imblue-dabadee committed:
fix: add cases for eth

mindofmar commented on src/components/simulation/PersonalSign.tsx:
```diff
@@ -0,0 +1,92 @@
+      case '1':
+        setChainLogoPath('/images/asset_logos/eth-mainnet.png');
+        break;
+      case '42161':
```
add `0xa4b1` and `0x1` to these cases

imblue-dabadee committed:
fix: handle hexadecimal chainIDs

imblue-dabadee committed:
chore: better naming for component

imblue-dabadee committed:
chore: more intuitive naming

imblue-dabadee committed:
fix: styling for buttons

mindofmar commented on src/components/simulation/PersonalSign.tsx:
```diff
@@ -0,0 +1,95 @@
+  // Fetch the metadata json
+  useEffect(() => {
+    const fetchData = async () => {
+      console.log("simulation.args.origin", simulation.args.origin)
```
remove this console log

mindofmar commented on src/components/simulation/PersonalSign.tsx:
```diff
@@ -0,0 +1,95 @@
+  useEffect(() => {
+    const fetchData = async () => {
+      console.log("simulation.args.origin", simulation.args.origin)
+      const response = await fetch(S3_URL_PROD + simulation.args.origin + '.json');
```
what happens if the `origin` doesn't have an object in S3?

imblue-dabadee committed:
config: update zip:chrome to incl package version

imblue-dabadee committed:
feat: logos for chains

imblue-dabadee committed:
chore: remove unnecessary divs

imblue-dabadee committed:
style: animations

imblue-dabadee committed:
feat: loading animations

imblue-dabadee committed:
fix: personal sign appears when loading

Should help with loading times since I assume it will now fetch for the PersonalSign assets while simulations are occurring

imblue-dabadee committed:
fix: handle !200s from CDN

imblue-dabadee committed:
fix: revert changes to confirmsimulationbutton

imblue-dabadee committed:
style: fix center on big screen for confirmbuttons

mindofmar commented on src/pages/popup.tsx:
```diff
@@ -98,6 +99,19 @@ const Popup = () => {
   }
 
+  // Personal Sign Screen
+  if (currentSimulation.args.method == SimulationMethodType.PersonalSign) {
```
Make this `===` please

mindofmar committed:
pr feedback

mindofmar committed:
rename s3 to cdn

mindofmar committed:
fix merge conflict after merging with develop
